src/typescript/frontend/src/app/trades/route.ts

```diff
@@ -0,0 +1,43 @@
+import { fetchSwapEvents } from "@/queries/market";
+import type { NextRequest } from "next/server";
+import { stringifyJSON } from "utils";
+
+type SwapSearchParams = {
+  marketID: string | null;
+  fromMarketNonce: string | null;
+};
+
+export type ValidSwapSearchParams = {
+  marketID: string;
+  fromMarketNonce: string;
+};
+
+const isNumber = (s: string) => !isNaN(parseInt(s));
```

Comment:
Let's not repeat this, use the other isNumber, and find the other one we use (I think I wrote a similar one somewhere, maybe like safeParseInt or something, it was for another route/fetcher/getter)